ProcessLib/HeatTransportBHE/BHE/BHE_1P.cpp

```diff
@@ -0,0 +1,140 @@
+/**
+ * \file
+ * \copyright
+ * Copyright (c) 2012-2020, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#include "BHE_1P.h"
+
+#include <boost/math/constants/constants.hpp>
+#include "FlowAndTemperatureControl.h"
+#include "Physics.h"
+#include "ThermoMechanicalFlowProperties.h"
+
+namespace ProcessLib
+{
+namespace HeatTransportBHE
+{
+namespace BHE
+{
+std::array<double, BHE_1P::number_of_unknowns> BHE_1P::pipeHeatCapacities()
+    const
+{
+    double const rho_r = refrigerant.density;
+    double const specific_heat_capacity = refrigerant.specific_heat_capacity;
+    double const rho_g = grout.rho_g;
+    double const porosity_g = grout.porosity_g;
+    double const heat_cap_g = grout.heat_cap_g;
+
+    return {{
+        /*pipe*/ rho_r * specific_heat_capacity,
+        /*grout*/ (1.0 - porosity_g) * rho_g * heat_cap_g,
+    }};
+}
+
+std::array<double, BHE_1P::number_of_unknowns> BHE_1P::pipeHeatConductions()
+    const
+{
+    double const lambda_r = refrigerant.thermal_conductivity;
+    double const rho_r = refrigerant.density;
+    double const Cp_r = refrigerant.specific_heat_capacity;
+    double const alpha_L = _pipe.longitudinal_dispersion_length;
+    double const porosity_g = grout.porosity_g;
+    double const lambda_g = grout.lambda_g;
+
+    // Here we calculate the laplace coefficients in the governing
+    // equations of the BHE.
+    return {{
+        // pipe, Eq. 19
+        (lambda_r + rho_r * Cp_r * alpha_L * _flow_velocity),
+        // grout, Eq. 21
+        (1.0 - porosity_g) * lambda_g,
+    }};
+}
+
+std::array<Eigen::Vector3d, BHE_1P::number_of_unknowns>
+BHE_1P::pipeAdvectionVectors(Eigen::Vector3d elem_direction_vec) const
+{
+    double const& rho_r = refrigerant.density;
+    double const& Cp_r = refrigerant.specific_heat_capacity;
+    Eigen::Vector3d adv_vector =
+        rho_r * Cp_r * _flow_velocity * elem_direction_vec;
+
+    return {// pipe, Eq. 19
+            adv_vector,
+            // grout, Eq. 21
+            {0, 0, 0}};
+}
+
+double BHE_1P::compute_R_gs(double const chi, double const R_g)
+{
+    return (1 - chi) * R_g;
+}
+
+void BHE_1P::updateHeatTransferCoefficients(double const flow_rate)
+
+{
+    auto const tm_flow_properties = calculateThermoMechanicalFlowPropertiesPipe(
+        _pipe.single_pipe, borehole_geometry.length, refrigerant, flow_rate);
+
+    _flow_velocity = tm_flow_properties.velocity;
+    _thermal_resistances =
+        calcThermalResistances(tm_flow_properties.nusselt_number);
+}
+
+/// Nu is the Nusselt number.
```

Comment:
```suggestion
// Nu is the Nusselt number.
```
Otherwise it is a documentation for the function.
You could do it, in the header though, with `\param Nu ...`

Reply:
Thanks, changed.